tests/common/quantization/data_generators.py

```diff
@@ -0,0 +1,372 @@
+"""
+ Copyright (c) 2023 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def get_quant_len_by_range(input_range: np.array, bits: int) -> np.array:
+    """Returns quant length for N-bit quantization by input range.
+
+    :param input_range: Input range of quantizer.
+    :param bits: Number of bits of quantization.
+
+    :return np.array: Length of quant.
+    """
+    return input_range.astype(np.float64) / (2**bits - 1)
+
+
+def get_quant_len(value_low: np.array, value_high: np.array, bits: int) -> np.array:
+    """Returns quant length for N-bit quantization by min and max values.
+
+    :param value_low: Low input value of quantizer.
+    :param value_high: High input value of quantizer.
+    :param bits: Number of bits of quantization.
+
+    :return np.array: Length of quant.
+    """
+    input_range = value_high.astype(np.float64) - value_low
+    return get_quant_len_by_range(input_range, bits)
+
+
+def get_quant_points(value_low: np.array, quant_len: np.array, bits: int) -> np.array:
+    """Returns array of quant point for quantization.
```

Comment:
Need better explanation as to what a `quant point` actually means. `"quant points" are floating-point levels to which the corresponding input floating point values will be brought to after application of fake quantization.`